src/plugin/nerd_plugin.js

```diff
@@ -97,37 +97,80 @@ class NerdPluginVirtual extends EventEmitter {
       settleIfOver: this.settleIfOver,
       initialBalance: opts.initialBalance
     })
+
     this.balance.on('_balanceChanged', (balance) => {
       this._log('balance changed to ' + balance)
       this.emit('_balanceChanged', balance)
-      this._sendBalance()
+      this.rpc.notify({ type: 'balance', balance: balance })
     })
+
     this.balance.on('under', (balance) => {
-      this._log('you should settle your balance of ' + balance)
-      this._sendSettle()
-    })
-    this.balance.on('over', (balance) => {
-      this._log('you should settle your balance of ' + balance)
-      this.emit('settlement', balance)
+      return co.wrap(this._settle).call(this, balance)
     })
 
-    if (this.settler) {
-      this.on('settlement', (balance) => {
-        if (!this.settleAddress) return
+    this.balance.on('over', (balance) => {
+      this._log('settling your balance of ' + balance)
+      if (!this.settleAddress || !this.settler) return
 
-        this.settler.send({
-          account: this.settleAddress,
-          amount: this._getSettleAmount(balance),
-          id: uuid()
-        })
+      this.settler.send({
+        account: this.settleAddress,
+        amount: this._getSettleAmount(balance),
+        id: uuid()
       })
+    })
 
+    if (this.settler) {
       this.settler.on('incoming_transfer', (transfer) => {
         if (transfer.account !== this.settleAddress) return
         this._log('received a settlement for ' + transfer.amount)
-        this._incomingSettle(transfer)
+        this.balance.add(transfer.amount).then(() => {
+          return this.balance.get()
+        })
       })
     }
+
+    this.rpc.addMethod('getPrefix', () => {
+      return this.getPrefix()
+    })
+
+    this.rpc.addMethod('getConnectors', () => {
+      return this.getConnectors()
+    })
+
+    this.rpc.addMethod('send', (transfer, settleAddress) => {
+      return co.wrap(this._handleSend).call(this, transfer, settleAddress)
+    })
+
+    this.rpc.addMethod('getBalance', () => {
+      return this.getBalance()
+    })
+
+    this.rpc.addMethod('getInfo', () => {
+      return this.getInfo()
+    })
+
+    this.rpc.addMethod('fulfillCondition', (transfer, fulfillment) => {
+      return co.wrap(this._fulfillConditionLocal).call(this, transfer, fulfillment)
+    })
+
+    this.rpc.addMethod('getFulfillment', (transferId) => {
+      return this.getFulfillment(transferId)
+    })
+
+    this.rpc.addMethod('rejectIncomingTransfer', (transferId, message) => {
+      return co.wrap(this._handleReject).call(this, transferId, message)
+    })
+
+    this.rpc.addMethod('replyToTransfer', (transferId, message) => {
+      return co.wrap(this._handleReply).call(this, transferId, message)
+    })
+
+    this.rpc.addMethod('settled', (transfer) => {
+      this._log('confirmed a settlement for ' + transfer.amount)
+      return this.balance.sub(transfer.amount).then(() => {
+        return Promise.resolve(true)
+      })
```

Comment:
`addMethod` is a fine pattern for RPC imo.

The alternative would be to create a class and register the class as the RPC handler, but then you're relying on the fact that the RPC layer correctly detects which methods are supposed to be exported, so this is a bit more verbose, but safer.